vk-video/src/parser/reference_manager.rs

```diff
@@ -34,12 +37,14 @@ pub struct ReferenceId(usize);
 pub(crate) struct ReferenceContext {
     pictures: ReferencePictures,
     next_reference_id: ReferenceId,
-    previous_frame_num: usize,
+    previous_frame_num: u16,
     prev_pic_order_cnt_msb: i32,
     prev_pic_order_cnt_lsb: i32,
     MaxLongTermFrameIdx: MaxLongTermFrameIdx,
     prevFrameNumOffset: i64,
     previous_picture_included_mmco_equal_5: bool,
+    is_missing_frames: bool,
+    allow_gaps_in_frames: bool,
```

Comment:
I'm not a fan of the naming here. `allow_gaps_in_frames` has the two problems that first, it sounds very similar to `gaps_in_frame_num_allowed`, which is an h264 flag describing a completely separate behavior, which we don't support, and second, is not very precise and could mean either that we allow some frames to be missing or that we allow chunks in the middle of a frame to be missing.

How about changing the first field to `detected_missed_frames`, and the second to `ignore_missed_frames`? Those names are still not as good as I would want them so I'm open to a discussion.

Also, I think we should change the names in the public API

Reply:
How about something like this?

```rs
enum MissedFrameHandling {
    /// When missed frames are detected, error on every subsequent frame that depends on them
    /// (i. e. fail on every frame until an IDR frame arrives)
    Strict,
    
    /// When missed frames are detected, try to decode later frames that depend on them anyway.
    /// This can produce decoded frames with very visible artifacts.
    Tolerant,
}
```